cpp/src/parquet/arrow/schema.cc

```diff
@@ -427,6 +428,12 @@ Status FieldToNode(const std::string& name, const std::shared_ptr<Field>& field,
     }
     case ArrowTypeId::EXTENSION: {
       auto ext_type = std::static_pointer_cast<::arrow::ExtensionType>(field->type());
+      // Built-in JSON extension is handled differently.
+      if (ext_type->extension_name() == std::string("arrow.json")) {
+        type = ParquetType::BYTE_ARRAY;
+        logical_type = LogicalType::JSON();
+        break;
```

Comment:
what does break means? should we return directly?

Reply:
Oh, I'm not sure how it got there. Removed.

Reply:
Reintroduced this break. It breaks `case` and is necessary.